shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -169,24 +126,27 @@ v8::MaybeLocal<v8::Value> PassValueToOtherContext(
   // the global handle at the right time.
   if (value->IsFunction()) {
     auto func = v8::Local<v8::Function>::Cast(value);
-    v8::Global<v8::Function> global_func(source_context->GetIsolate(), func);
-    v8::Global<v8::Context> global_source(source_context->GetIsolate(),
-                                          source_context);
 
-    size_t func_id = store->take_func_id();
-    store->functions()[func_id] =
-        std::make_tuple(std::move(global_func), std::move(global_source));
     {
       v8::Context::Scope destination_scope(destination_context);
-      v8::Local<v8::Value> proxy_func = gin_helper::CallbackToV8Leaked(
-          destination_context->GetIsolate(),
-          base::BindRepeating(&ProxyFunctionWrapper, store, func_id,
-                              support_dynamic_properties));
-      FunctionLifeMonitor::BindTo(destination_context->GetIsolate(),
-                                  v8::Local<v8::Object>::Cast(proxy_func),
-                                  store->GetWeakPtr(), func_id);
-      object_cache->CacheProxiedObject(value, proxy_func);
-      return v8::MaybeLocal<v8::Value>(proxy_func);
+      gin_helper::Dictionary state = gin_helper::Dictionary::CreateEmpty(
+          destination_context->GetIsolate());
+      DCHECK(IsTrue(state.GetHandle()->SetPrivate(
+          destination_context,
+          v8::Private::ForApi(destination_context->GetIsolate(),
+                              gin::StringToV8(destination_context->GetIsolate(),
+                                              "electron_ref")),
```

Comment:
Does this still need to be set using `v8::Private` now that it's being passed as `data` to the function callback rather than being attached to the function? Could it simply be a key in the dict?

Reply:
@nornagon I'm not entirely sure whether `data` is completely safe whereas v8 privates have certain guarantees and not too much overhead.

Reply:
Actually, I remember thinking about this, hard no.  The object prototypes in the target context could be overridden to intercept the cross-context function if we don't use a private.  The private here protects us against context-side prototype pollution